lib/spack/spack/spec.py

```diff
@@ -2650,7 +2650,7 @@ def write(s, c):
                         a_str = ' arch' + c + str(self.architecture) + ' '
                         write(fmt % (a_str), c)
                 elif c == '/':
-                    out.write('/' + fmt % (self.dag_hash(7)))
+                    out.write('-' + fmt % (self.dag_hash(7)))
```

Comment:
I have removed this change since it may no longer be needed. If it a test determines later it is needed then it will be changed in a separate pull request.